src/main/java/org/broadinstitute/hellbender/tools/walkers/sv/SVAnnotateEngine.java

```diff
@@ -672,17 +843,22 @@ protected Map<String, Object> annotateStructuralVariant(final VariantContext var
             }
         }
 
+        // for CPX events, update SV segments to annotate nearest TSS
+        if (overallSVType == GATKSVVCFConstants.StructuralVariantAnnotationType.CPX) {
+            svSegments = getSegmentForNearestTSS(svSegments, complexType);
+        }
+
         // annotate nearest TSS for intergenic variants with no promoter overlaps
         if (gtfIntervalTrees != null && gtfIntervalTrees.getTranscriptionStartSiteTree() != null &&
-                !variantConsequenceDict.containsKey(GATKSVVCFConstants.PROMOTER) && noCodingAnnotations) {
+                !variantConsequenceDict.containsKey(GATKSVVCFConstants.PROMOTER) && isIntergenic) {
```

Comment:
You're right that I could call isIntergenic() here for the same result; however, a variant can be intergenic and still have promoter annotations, and in that case I think it's confusing to reuse the isIntergenic() function to identify a different set of criteria. And I think I need the isIntergenic boolean because I need to store the information about whether there are any coding annotations to use later to apply the INTERGENIC flag. Alternatively I could modify the function to ignore the noncoding annotations, and maybe create two different functions - one that ignores all noncoding annotations and one that ignores noncoding annotations except for promoters.

Reply:
Updated isIntergenic() to check for an absence of protein-coding consequences and ignore other consequences, so that the result is valid as long as all protein-coding consequences have been annotated, even if non-coding annotations have been added. Let me know if that addresses the issue